### PR TITLE
Cleaner vector code structure in feature transformer

### PR DESCRIPTION
**Pull request by snicolet**
No functional change

### CODE REVIEW

File: src/nnue/nnue_feature_transformer.h [hunk 1]
```diff
@@ -182,115 +182,134 @@ namespace Stockfish::Eval::NNUE {
...
       const __m512i Control = _mm512_setr_epi64(0, 2, 4, 6, 1, 3, 5, 7);
       const __m512i Zero = _mm512_setzero_si512();
 
+      for (IndexType p = 0; p < 2; ++p) 
+      {
+          const IndexType offset = HalfDimensions * p;
+          auto out = reinterpret_cast<__m512i*>(&output[offset]);
+          for (IndexType j = 0; j < NumChunks; ++j) 
+          {
+              __m512i sum0 = _mm512_load_si512(&reinterpret_cast<const __m512i*>(accumulation[perspectives[p]])[j * 2 + 0]);
```
Comment (Sopel97) [inline src/nnue/nnue_feature_transformer.h:199]: very long lines

**Comment by snicolet** [comment]
I have moved the accumulator pointers outside the inner loops, which allows for shorter lines.

For gcc-10 and the AVX2 path this does not the binary, as the compiler was doing the same "optimization" already, but for other compiler/target combo this may change the emitted assembly.

Test for non-regression ongoing on fishtest: https://tests.stockfishchess.org/tests/view/60b69888457376eb8bca9fc6
(not looking good after 6700 games)

If the test fails I will cut the long lines manually in the C++ code and try again.

**Comment by snicolet** [comment]
So, second non-regression test: https://tests.stockfishchess.org/tests/view/60b96b39457376eb8bcaa26e

**Comment by snicolet** [comment]
Merged via 8f081c86f7f8827ea35fc687e6f6591950cc8f90.

**Pull request closed without merging**